drivers/include/lm75.h

```diff
@@ -0,0 +1,246 @@
+ * @param[in] temperature        buffer where temperature value will be written
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ */
+int lm75_get_temperature_raw(lm75_t *dev, int *temperature);
+
+/**
+ * @brief Temperature values of LM75 sensor
+ *
+ * Gets the device's temperature register with the lm75_get_temperature_raw
+ * function and then returns the values in mºC, truncating values smaller than this, if available
+ *
+ * @param[in] dev                device structure
+ * @param[in] temperature        buffer where temperature value will be written in mºC
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR, on error
+ */
+int lm75_get_temperature(lm75_t *dev, int *temperature);
+
+/**
+ * @brief Sets the values for Overtemperature shutdown(OS) and Hysteresis temperature(HYST).
+ * OS gives the temperature's higher bound and HYST the lower bound
+ * values are rounded to the lowest value that the device supports,
+ *
+ * @param[in] dev           device structure
+ * @param[in] temp_os       desired OS temperature in mºC
+ * @param[in] temp_hyst     desired HYST temperature in mºC
+ * @param[in] cb            callback that is called from interrupt context
+ * @param[in] *arg          optional arguments for the gpio_init_int function
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ * @return LM75_ERROR, on temperature setting related error
+ */
+int lm75_set_temp_limits(lm75_t *dev, int temp_hyst, int temp_os, gpio_cb_t cb, void *arg);
+
+/**
+ * @brief Overshutdown temperature value of LM75 sensor
+ *
+ * Reads the sensor OS temperature value from TOS_REG in ºC.
+ *
+ * @param[in] dev                 device structure
+ * @param[out] temperature        buffer where OS temperature value will be written
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ */
+int lm75_get_os_temp(lm75_t *dev, int *temperature);
+
+/**
+ * @brief Hysteresis temperature value of LM75 sensor
+ *
+ * Reads the sensor hysteresis temperature value from THYST_REG in ºC.
+ *
+ * @param[in] dev                 device structure
+ * @param[out] temperature        buffer where HYST temperature value will be written
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ */
+int lm75_get_hyst_temp(lm75_t *dev, int *temperature);
+
+/**
+ * @brief Read the current state of the OS pin to see if it's active.
+ *
+ * Read the configuration register to see the OS pin's polarity and
+ * then reads its state. Then outputs if the pin is active and whether
+ * it's in the low and active or high and active.
+ *
+ * @param[in] dev               device structure
+ * @param[out] os_pin_state     pointer to the state of the OS pin - 0 for inactive and 1 for active
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR, on pin reading error
+ */
+int lm75_get_os_pin(lm75_t *dev, uint8_t *os_pin_state);
```

Comment:
```suggestion
int lm75_get_os_pin(lm75_t *dev, bool *os_pin_state);
```

if you want to do it that way you might as well use a boolean